src/ripple/rpc/handlers/ServerInfo.cpp

```diff
@@ -22,13 +22,298 @@
 #include <ripple/app/reporting/P2pProxy.h>
 #include <ripple/json/json_value.h>
 #include <ripple/net/RPCErr.h>
+#include <ripple/protocol/LedgerFormats.h>
+#include <ripple/protocol/SField.h>
+#include <ripple/protocol/TER.h>
+#include <ripple/protocol/TxFormats.h>
+#include <ripple/protocol/digest.h>
 #include <ripple/protocol/jss.h>
 #include <ripple/rpc/Context.h>
 #include <ripple/rpc/Role.h>
-#include <ripple/rpc/impl/TransactionSign.h>
+#include <boost/algorithm/string.hpp>
+#include <sstream>
 
 namespace ripple {
 
+class Definitions
+{
+private:
+    std::string
+    translate(std::string inp)
+    {
+        auto replace = [&](const char* f, const char* r) -> std::string {
+            std::string out = inp;
+            boost::replace_all(out, f, r);
+            return out;
+        };
+
+        auto find = [&](const char* s) -> bool {
+            return inp.find(s) != std::string::npos;
+        };
+
+        if (find("UINT"))
+        {
+            if (find("256") || find("160") || find("128"))
+                return replace("UINT", "Hash");
+            else
+                return replace("UINT", "UInt");
+        }
+
+        if (inp == "OBJECT")
+            return "STObject";
+        if (inp == "ARRAY")
+            return "STArray";
+        if (inp == "AMM")
+            return "AMM";
+        if (inp == "ACCOUNT")
+            return "AccountID";
+        if (inp == "LEDGERENTRY")
+            return "LedgerEntry";
+        if (inp == "NOTPRESENT")
+            return "NotPresent";
+        if (inp == "PATHSET")
+            return "PathSet";
+        if (inp == "VL")
+            return "Blob";
+        if (inp == "DIR_NODE")
+            return "DirectoryNode";
+        if (inp == "PAYCHAN")
+            return "PayChannel";
+        if (inp == "XCHAIN_BRIDGE")
+            return "XChainBridge";
+
+        std::string out;
+        size_t pos = 0;
+        for (;;)
+        {
+            pos = inp.find("_");
+            if (pos == std::string::npos)
+                pos = inp.size();
+            std::string token = inp.substr(0, pos);
+            if (token.size() > 1)
+            {
+                boost::algorithm::to_lower(token);
+                token.data()[0] -= ('a' - 'A');
+                out += token;
+            }
+            else
+                out += token;
+            if (pos == inp.size())
+                break;
+            inp = inp.substr(pos + 1);
+        }
+        return out;
+    };
+
+    Json::Value
+    generate()
+    {
+        Json::Value ret{Json::objectValue};
+        ret[jss::TYPES] = Json::objectValue;
+
+        ret[jss::TYPES]["Done"] = -1;
+        std::map<int32_t, std::string> type_map{{-1, "Done"}};
+        for (const auto& pair : sTypeMap)
```

Comment:
Consider using a structured binding here. Also, the `const` goes after the `auto`. I.e.